### PR TITLE
Feature-28: add removeTechnicians function to the techniciansController file

### PR DESCRIPTION
**Pull request by ValentinSilvestri**
Added the removeTechnicians by id function to the techniciansController file.

### CODE REVIEW

File: src/controllers/techniciansController.js [hunk 1]
```diff
@@ -17,8 +18,24 @@ const getTechniciansBy = (name, lastName) => {
...
+const removeTechniciansBy = id => {
+    const search = technicians.find(technician => technician.id.toString() === id);
+    const updatedTechnicians = technicians.filter(technician => !(technician.id.toString() === id));
+    if (search){
+        fs.writeFileSync(
+            __dirname + "/../data/technicians.json",
+            JSON.stringify(updatedTechnicians),
+            { encoding: 'utf8', flag: 'w' }
+        );
+        return ("Technician with id " + id + " deleted.").toString();
```
Comment (lmalvasia) [inline src/controllers/techniciansController.js:30]: the toString() is not necessary. You can do:

```suggestion
        return `Technician with id ${id} deleted`;
```

File: src/controllers/techniciansController.js [hunk 2]
```diff
@@ -17,8 +18,24 @@ const getTechniciansBy = (name, lastName) => {
...
+    } else {
+        return ("Technician with id " + id + " not found.").toString();
```
Comment (lmalvasia) [inline src/controllers/techniciansController.js:32]: The same that above.
the toString() is not necessary. You can do:
return `Technician with id ${id} not found.`;
```suggestion
        return `Technician with id ${id} not found.`;
```

File: src/index.js [hunk 3]
```diff
@@ -27,10 +27,14 @@ app.get("/getAllBoylers", (req, res) => {
   res.json(boylers);
 });
 
-app.get("/getAllBoylerTypesByDescription/:description", (req, res) => {
-  const description = req.params.description;
-  const boylerTyperByDescription = boylerTypesController.getBoylerTypesByDescription(description);
-  res.json(boylerTyperByDescription);
+app.get("/getTechniciansById/:id", (req, res) => {
+  const id = req.params.id.toString();
```
Comment (lmalvasia) [inline src/index.js:31]: All the path params are strings, it's not necessary this conversion.

File: src/index.js [hunk 4]
```diff
@@ -27,10 +27,14 @@ app.get("/getAllBoylers", (req, res) => {
   res.json(boylers);
 });
 
-app.get("/getAllBoylerTypesByDescription/:description", (req, res) => {
```
Comment (lmalvasia) [inline src/index.js:30]: Why did you remove this part? 🤔 Could it be a bad rebase? 🤔

File: src/index.js [hunk 5]
```diff
@@ -77,6 +81,12 @@ app.get("/getTechniciansBy/:name/:lastName", (req, res) => {
   res.json(technician);
 });
 
+app.get("/removeTechniciansBy/:id", (req, res) => {
+  const id = req.params.id.toString();
```
Comment (lmalvasia) [inline src/index.js:85]: It's not necessary this conversion

**Review by lmalvasia — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 4 above)
Comment (ValentinSilvestri) [reply src/index.js:30]: That's why, probably in a rebase a conflict occurred and, instead of accepting the changes, I rewrite part of the code and mess it up.

**Review by lmalvasia — APPROVED** [review]
